tests/integration_test.py

```diff
@@ -0,0 +1,309 @@
+import requests_mock
+import io
+import os
+
+from tests.utils.base_test_case import BaseTestCase
+from tests.utils import mock_polyswarm_api_results
+from tests.utils import file_utils
+
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
+
+
+@requests_mock.Mocker()
+class IntegrationTest(BaseTestCase):
+    """
+        These tests mock the resource end-point, thus also sanity testing polyswarm-api.
```

Comment:
We should not be mocking http requests in `polyswarm-cli`, we should mock responses from `polyswarm-api` only. This creates a layer of abstraction that automatically deals with changes in the http api. As it is, if we change something in the responses from the server, we have to come and fix tests both in `cli` and `api` instead of changing only `api` even though we guarantee that external methods still have the same behavior. This should be completely transparent for people using the programmatic interface in `api`. These tests should be included in the `api`, and it should break there, not here. As long as `api` returns what we expect, the `cli` should work. This is my opinion, though, and I would like @supernothing comments on this.

Reply:
In unit tests makes sense to mock the `polyswarm-api`, but mocking the resource API is the way to go for integration tests. They were already useful: The unit tests for "download" for instance were buggy (the mocked results didn't correspond to the actual results returned by `api.download()`. The integration tests don't mean to be exhaustive, they only check all the major operations for sanity check sake.